src/main/java/gui/view_model/MonthlyCalendarViewModel.java

```diff
@@ -1,38 +1,120 @@
 package gui.view_model;
 
+import com.calendarfx.model.CalendarEvent;
 import com.calendarfx.model.Entry;
 import data_gateway.event.EventReader;
 import data_gateway.event.ObservableEventRepository;
 import gui.utility.EventHelper;
 import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
 import javafx.collections.ObservableList;
 
-import java.util.List;
+import java.io.IOException;
+import java.util.*;
+
 
 public class MonthlyCalendarViewModel extends ViewModel {
 
     private final ObservableEventRepository repository;
 
     private final ObservableList<Entry<String>> entryList;
 
+    private final Map<String, Long> entryToEventIdMapping;
+
     public MonthlyCalendarViewModel(ObservableEventRepository repository) {
         this.repository = repository;
         repository.addCreationObserver(this::handleCreation);
         repository.addUpdateObserver(this::handleUpdate);
 
-        List<EventReader> eventReaderList = this.repository.getAllEvents();
-        this.entryList = FXCollections.observableArrayList(EventHelper.eventReaderToEntry(eventReaderList));
+        this.entryList = FXCollections.observableArrayList(new ArrayList<>());
+        this.entryToEventIdMapping = new HashMap<>();
+        initializeIdMappingAndEntryList(this.repository.getAllEvents());
+
+        this.entryList.addListener((ListChangeListener<Entry<String>>) c -> {
+            while (c.next()) {
+                if (c.wasAdded()) {
+                    Entry<String> entry = c.getAddedSubList().get(0);
+
+                    // if entryList and entryToEventIdMapping has the same size,
+                    // entry added to entryList is due to changes in data in the
+                    // repository and can be ignored as the mapping has been added.
+
+                    if (entryList.size() == entryToEventIdMapping.size() + 1) {
```

Comment:
I think this only gets updated when an event is created, not when it's attributes are changed, is there a way to fix this?